### PR TITLE
Crash protection for VM

### PR DESCRIPTION
**Pull request by junmendoza**
1. Set HasRunWithoutCrash flag to true only if no exception/errors
occured
2. Remove redundant error handler

@pboyer @Benglin 
As discussed, ive put in a fix to set the flag HasRunWithoutCrash only if no errors occurred. Pls see if this is sufficient.

My other concern is, I can get a dyn file that crashes the VM and add it as a CrashProtection test. But if the dyn file issue happens to be fixed, then the CrashProtection test will regress because it will always assume the test to throw a VM exception.

Do you have any suggestion how properly test this?
Im wary of adding code that forces the VM to crash

### CODE REVIEW

File: src/DynamoCore/Properties/Resources.en-US.resx [hunk 1]
```diff
@@ -233,7 +233,7 @@
   <data name="CannotOpenFile" xml:space="preserve">
```
Comment (Benglin) [inline src/DynamoCore/Properties/Resources.en-US.resx]: Hi @junmendoza, just want to make sure you modified `Resources.resx`, then overwrite `Resources.en-US.resx` with `Resources.resx`? Just trying to ensure these files are completely identical.

Comment (junmendoza) [inline src/DynamoCore/Properties/Resources.en-US.resx]: @Benglin yes the changes are identical, although i ddnt overwrite Resources.en-US.resx. I just made sure the changes were the same. Is it recommended to always overwrite Resources.en-US.resx if changes are made to Resources.resx

File: src/DynamoCore/Models/DynamoModel.cs [hunk 2]
```diff
@@ -1669,7 +1669,7 @@ private TaskDialogEventArgs DisplayEngineFailureMessage(Exception exception)
 
             string description = (exception is HeapCorruptionException)
                 ? exception.Message
-                : @Properties.Resources.ExceptionIsNotHeapCorruptionDescription;
+                : @Properties.Resources.DisplayEngineFailureMessageDescription;
 
             const string imageUri = "/DynamoCoreWpf;component/UI/Images/task_dialog_crash.png";
             var args = new TaskDialogEventArgs(
```
Comment (Benglin) [inline src/DynamoCore/Models/DynamoModel.cs]: Ouch, a `DynamoCoreWpf` reference in `DynamoModel`! I have done the following:

- Removed `DynamoCoreWpf` strings from `DynamoCore` project [in this pull request](https://github.com/DynamoDS/Dynamo/pull/4398).
- Added [a new task](http://adsk-oss.myjetbrains.com/youtrack/issue/MAGN-7308) for refactoring `TaskDialogEventArgs`.

Please help to review the above PR :)

File: src/DynamoCore/Models/HomeWorkspaceModel.cs [hunk 3]
```diff
@@ -440,10 +440,14 @@ public virtual void OnEvaluationStarted(EventArgs e)
         public event EventHandler<EvaluationCompletedEventArgs> EvaluationCompleted;
         public virtual void OnEvaluationCompleted(EvaluationCompletedEventArgs e)
         {
-            this.HasRunWithoutCrash = true;
+            if(e.EvaluationSucceeded)
+            {
+                this.HasRunWithoutCrash = true;
+            }
```
Comment (Benglin) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: I'm not sure what are the odds of `OnEvaluationStarted` not getting called before `OnEvaluationCompleted`. If that ever happens then `HasRunWithoutCrash` will be left with an unknown value. Should we instead do this?

```csharp
// Assignment without condition.
this.HasRunWithoutCrash = e.EvaluationSucceeded;
```

Comment (junmendoza) [inline src/DynamoCore/Models/HomeWorkspaceModel.cs]: yea good point

**Comment by Benglin** [comment]
LGTM!

**Pull request merged**